From 94e73f1953f44f786a5d1f83283a886adc5530ab Mon Sep 17 00:00:00 2001
From: Marco Martin <notmart@gmail.com>
Date: Mon, 1 Mar 2021 18:31:48 +0100
Subject: [PATCH] break the size bindings

Explicitly break the binding as we need this to be set only at startup.
if the bindings are active, after this the window is resized by the
compositor and then the bindings are reevaluated, then the window
size would reset ignoring what the compositor asked.
---
 src/controls/AbstractApplicationWindow.qml | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/controls/AbstractApplicationWindow.qml b/src/controls/AbstractApplicationWindow.qml
index 059b6f67..8b7b7847 100644
--- a/src/controls/AbstractApplicationWindow.qml
+++ b/src/controls/AbstractApplicationWindow.qml
@@ -317,6 +317,14 @@ QQC2.ApplicationWindow {
     height: Settings.isMobile ? Units.gridUnit * 45 : Units.gridUnit * 40
     visible: true
 
+    Component.onCompleted: {
+        // Explicitly break the binding as we need this to be set only at startup.
+        // if the bindings are active, after this the window is resized by the
+        // compositor and then the bindings are reevaluated, then the window
+        // size would reset ignoring what the compositor asked.
+        root.width = root.width;
+        root.height = root.height;
+    }
 
     QtObject {
         id: internal
-- 
GitLab

